cli/slsa-verifier/main_regression_test.go

```diff
@@ -1526,12 +1526,57 @@ func Test_runVerifyGithubAttestation(t *testing.T) {
 	bcrPublisherBuilderID := "https://github.com/bazel-contrib/publish-to-bcr/.github/workflows/publish.yaml"
 
 	tests := []struct {
-		name      string
-		artifact  string
-		source    string
-		builderID string
-		err       error
+		name             string
+		artifact         string
+		source           string
+		sourceTag        *string
+		sourceVersionTag *string
+		builderID        string
+		err              error
 	}{
+		{
+			name:      "module.bazel using publishing builder",
+			artifact:  "MODULE.bazel",
+			source:    "github.com/aspect-build/rules_lint",
+			builderID: bcrPublisherBuilderID,
+		},
+		{
+			name:      "module.bazel using publishing builder and source tag",
+			artifact:  "MODULE-on-tag.bazel",
+			source:    "github.com/aspect-build/rules_lint",
+			sourceTag: pString("v1.3.4"),
+			builderID: bcrPublisherBuilderID,
+		},
+		{
+			name:      "module.bazel using publishing builder and incorrect source tag",
+			artifact:  "MODULE-on-tag.bazel",
+			source:    "github.com/aspect-build/rules_lint",
+			sourceTag: pString("v1.3.5"),
+			builderID: bcrPublisherBuilderID,
+			err:       serrors.ErrorMismatchTag,
+		},
+		{
+			name:             "module.bazel using publishing builder and source versioned tag",
+			artifact:         "MODULE-on-tag.bazel",
+			source:           "github.com/aspect-build/rules_lint",
+			sourceVersionTag: pString("v1.3.4"),
+			builderID:        bcrPublisherBuilderID,
+		},
+		{
+			name:             "module.bazel using publishing builder and partial source versioned tag",
+			artifact:         "MODULE-on-tag.bazel",
+			source:           "github.com/aspect-build/rules_lint",
+			sourceVersionTag: pString("v1.3"),
```

Comment:
I realize there aren't any tests that check for tags that are pre-releases. But can you add a couple, so we can be sure about the behavior.

Reply:
hrmm.. okay lemme try to generate one